src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -260,8 +260,10 @@ public virtual void Contextualize(ViewContext viewContext)
         /// <inheritdoc />
         public IHtmlContent AntiForgeryToken()
         {
-            var html = _htmlGenerator.GenerateAntiforgery(ViewContext);
-            return html ?? HtmlString.Empty;
+            var content = new HtmlContentBuilder();
+            _htmlGenerator.GenerateAntiforgery(ViewContext, content);
+
+            return content;
```

Comment:
The tag helpers have a builder pattern -- get a `TagHelperOutput` and muck w/ it. The HTML helpers mostly have a factory pattern -- they return `IHtmlContent`s. Neither is ideal for `IHtmlGenerator` but that's not really relevant here since this method is unusual and can't return a `TagBuilder`.

In any case either pattern works. I prefer this pattern here because it doesn't force an unnecessary adapter in the tag helper case. Since it doesn't match the rest of `IHtmlGenerator` and you disagree, I'll switch.